src/languages/javascript.js

```diff
@@ -65,10 +66,18 @@ export default function(hljs) {
       // `<blah />` (self-closing)
       // handled by simpleSelfClosing rule
 
-      // `<From extends string>`
-      // technically this could be HTML, but it smells like a type
       let m;
       const afterMatch = match.input.substring(afterMatchIndex);
+
+      // some more template typing stuff
+      //  <T = any>(key?: string) => Modify<
+      if ((m = afterMatch.match(/^\s+=/))) {
```

Comment:
```suggestion
      if ((m = afterMatch.match(/^\s*=/))) {
```

Personally, I've never seen anyone syntactically leave out spaces in generics but it is valid TS to not have spaces in between the `=` and also invalid JSX to not have an attribute on the LHS of an `=`.

The above change still keeps highlighting and unit tests pass.

![image](https://user-images.githubusercontent.com/1246453/196017313-494502ca-d9b6-4344-83b9-a6d786412be8.png)